app/controllers/pdl.py

```diff
@@ -1206,12 +1222,77 @@ def generate_title_i_total_state_distribution_response(title_id, start_year, end
     for state in sorted_summary:
         state['totalPaymentInDollars'] = round(state['totalPaymentInDollars'], 2)
 
-    sorted_data_by_year['2014-2021'] = sorted_summary
+    sorted_data_by_year[str(start_year) + '-' + str(end_year)] = sorted_summary
 
     result_dict = dict(sorted_data_by_year)
 
     return result_dict
 
+def generate_title_ii_total_state_distribution_response(title_id, start_year, end_year):
+    session = Session()
+
+    # construct the query
+    program_query = (
+        session.query(
+            Payment.state_code.label('state'),
+            Payment.year.label('year'),
+            Payment.payment.label('totalPaymentInDollars')
+        ).filter(
+            Payment.title_id == title_id,
+            Payment.year.between(start_year, end_year),
+            # when calculating the title level summaries, only calculate Total CRP (sub_program_id == 102) from CRP (program_id == 108)
```

Comment:
Noting one small thing that these program IDs could change based on how the data gets ingested. So, it may not be highly relevant. The other details are good and essential. Thanks.